extension/js/common/core/crypto/pgp/openpgp-key.ts

```diff
@@ -118,74 +113,67 @@ export class OpenPGPKey {
     if (encryptedPacketCount) {
       throw new Error(`Cannot encrypt a key that has ${encryptedPacketCount} of ${secretPackets.length} private packets still encrypted`);
     }
-    await prv.encrypt(passphrase);
-    if (!prv.isFullyEncrypted()) {
+    const encryptedPrv = await opgp.encryptKey({ privateKey: prv, passphrase });
+    if (!OpenPGPKey.isFullyEncrypted(encryptedPrv)) {
       throw new Error('Expected key to be fully encrypted after prv.encrypt');
     }
-    await OpenPGPKey.convertExternalLibraryObjToKey(prv, key);
+    await OpenPGPKey.convertExternalLibraryObjToKey(encryptedPrv, key);
   };
 
-  public static decryptMessage = async (message: OpenPGP.message.Message, privateKeys: Key[], passwords?: string[]) => {
+  public static decryptMessage = async (message: OpenPGP.Message<OpenPGP.Data>, privateKeys: Key[], passwords?: string[]) => {
+    const opgpKeys = await Promise.all(privateKeys.map(key => OpenPGPKey.extractExternalLibraryObjFromKey(key)));
     return await message.decrypt(
-      privateKeys.map(key => OpenPGPKey.extractExternalLibraryObjFromKey(key)),
-      passwords,
-      undefined,
-      false
+      opgpKeys.filter(key => key.isPrivate()).map(key => key as OpenPGP.PrivateKey),
+      passwords
     );
   };
 
   public static encryptMessage: PgpMsgMethod.Encrypt = async ({ pubkeys, signingPrv, pwd, data, filename, armor, date }) => {
-    const message = opgp.message.fromBinary(data, filename, date);
-    const options: OpenPGP.EncryptOptions = { armor, message, date };
-    if (pubkeys) {
-      options.publicKeys = [];
-      for (const pubkey of pubkeys) {
-        const { keys: publicKeys } = await opgp.key.readArmored(KeyUtil.armor(pubkey));
-        options.publicKeys.push(...publicKeys);
-        // TODO: Investigate why unwrapping doesn't work - probably the object
-        // came from the background page so it wasn't properly deserialized
-        // options.publicKeys.push(OpenPGPKey.unwrap(pubkey));
-      }
-    }
-    if (pwd) {
-      options.passwords = [pwd];
-    }
     if (!pubkeys && !pwd) {
       throw new Error('no-pubkeys-no-challenge');
     }
+    const message = await opgp.createMessage({ binary: data, filename, date });
+    const encryptionKeys = await Promise.all(pubkeys?.map(OpenPGPKey.extractExternalLibraryObjFromKey) ?? []);
+    // TODO: Investigate unwrapping?
```

Comment:
Highlighting, although unsure about what the todo means

Reply:
Perhaps, it's outdated.
The keys coming from the background are now `wrapped` (packed) as of this PR